examples/data-objects/shared-text/src/document.ts

```diff
@@ -13,33 +13,27 @@ const rootMapId = "root";
  * A document is a collection of collaborative types.
  */
 export class Document {
-    public static async load(runtime: IFluidDataStoreRuntime): Promise<Document> {
+    public static async load(runtime: IFluidDataStoreRuntime, existing: boolean): Promise<Document> {
         let root: ISharedMap;
 
-        if (!runtime.existing) {
+        if (!existing) {
             root = SharedMap.create(runtime, rootMapId);
             root.bindToContext();
         } else {
             root = await runtime.getChannel(rootMapId) as ISharedMap;
         }
 
-        const document = new Document(runtime, root);
-
-        return document;
-    }
-
-    /**
-     * Flag indicating whether the document already existed at the time of load
-     */
-    public get existing(): boolean {
-        return this.runtime.existing;
+        return new Document(runtime, root, existing);
     }
 
     /**
      * Constructs a new document from the provided details
      */
-    private constructor(public runtime: IFluidDataStoreRuntime, private readonly root: ISharedMap) {
-    }
+     private constructor(
+        public runtime: IFluidDataStoreRuntime,
+        private readonly root: ISharedMap,
+        public readonly existing: boolean,
```

Comment:
Is there a way to not have existing on this class?
It's ok as is (as we push it out from runtime into these leafy cases), but it would be even better if it was gone here as well

Reply:
There are some references to it. Let me see if I can do something about it..